### DISCUSSION TITLE
Multiple selected item in list

### DISCUSSION THREAD

**Discussion by Clashkid155** [charmbracelet/bubbles#352] [Q&A] [answered]
I can't seem to find any way to indicate when multiple items are selected in a list. I'm more concerned about the styling when multiple items are selected for the actual selection, I could just store the index when each item is selected.

Cutting all the long talk, can I change the style of each item individually?

**Comment by jon4hz** [comment, marked as answer]
You can write a [custom delegate function](https://github.com/charmbracelet/bubbles/blob/master/list/list.go#L38) for your list, which applies different styles on the items according to their "selection status".

**Reply by Clashkid155** [reply]
The selection status available is basically selected, dimmed and i think filtered if I'm not mistaken. I want something like a checkbox or how selecting multiple items highlights them on our various file explorer.

Well, I would appreciate an example of your suggestion.

**Reply by jon4hz** [reply]
Here's an example of a very basic, customized list delegation: https://gist.github.com/jon4hz/413571f10e3e85ca0a2e413d2bddbc0a
This allows you to fully customize all the table items. 
If you don't require a lot of customization regarding the items of the list itself, you could also implement a custom [DefaultItem](https://github.com/charmbracelet/bubbles/blob/master/list/defaultitem.go#L65) and/or override the styles of the default delegation (to change coloring, etc), which is the simpler solution but allows less customization :)

**Reply by Clashkid155** [reply]
Thanks for this!! I'll get back to you once i try it out

**Reply by Clashkid155** [reply]
Thanks for your example!!
I added the below code to change the selection status which works, but I would like to know if there's a better way.

```go
case tea.KeyMsg:
		switch msg.String() {
		case "enter":
			lis := m.list.SelectedItem().(listItem)
			lis.triggered = !lis.triggered
			m.list.SetItem(m.list.Index(), lis)
		}
```
Visual demo:
![demo](https://user-images.githubusercontent.com/21040700/225618832-2e911f37-9d7a-4212-b0ee-2b55ea5b9d81.gif)

**Reply by jon4hz** [reply]
Nice! Glad I could help :D

Also, make sure to checkout the `Update` method of the delegation. 
https://gist.github.com/jon4hz/413571f10e3e85ca0a2e413d2bddbc0a#file-checkbox-list-go-L102

Method description: https://github.com/charmbracelet/bubbles/blob/db8de3bf530c16f884d49915eed3237bb9c8eb96/list/list.go#L48-L52

I'm not a 100% sure but this is probably the right way to prevent interference with filtering.

**Reply by Clashkid155** [reply]
I have been checking it out but I can't seem to figure out how to use it in my case.
An example I have been studying is [Here](https://github.com/charmbracelet/bubbletea/blob/248eb83001a7ab6a11c80534bd4e3e96a6185f4a/examples/list-fancy/delegate.go#L9). Unlike this example I need to change the value of triggered as per your example.

> ```go
> case tea.KeyMsg:
> 		switch msg.String() {
> 		case "enter":
> 			lis := m.list.SelectedItem().(listItem)
> 			lis.triggered = !lis.triggered
> 			m.list.SetItem(m.list.Index(), lis)
> 		}
> ```
> 
Although this works, it feels hacky and unsafe since I'm only changing the value of `triggered` then replacing the previous list item with the new one.

I'm currently using your example as my test ground and trying different methods with it.
